connection.go

```diff
@@ -633,7 +633,6 @@ func (c *Connection) openChannel() (*Channel, error) {
 // this connection.
 func (c *Connection) closeChannel(ch *Channel, e *Error) {
 	ch.shutdown(e)
-	c.releaseChannel(ch.id)
```

Comment:
If we do this documentation for a few methods needs updating. I'm looking into if we can adopted.